### DISCUSSION TITLE
Grains longAxis histogram

### DISCUSSION THREAD

**Discussion by lucapbelotti** [mtex-toolbox/mtex#1086] [Ask Anything] [answered]
Hi everyone,

I am trying to study the shape preferred orientation of the grains using the rose diagrams as shown in (https://mtex-toolbox.github.io/EllipseBasedParameters.html). However, I get the following error for the histogram function:
`
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd,'threshold',[HAGB*degree LAGB*degree]);
[omega,a,b] = grains.fitEllipse;
weights = grains.area .* (grains.aspectRatio-1);
figure
histogram(grains.longAxis,50, 'weights', weights)
`
![image](https://user-images.githubusercontent.com/41550623/121358141-a8a70d00-c932-11eb-8c9a-0df29b2905e3.png)

Am I missing anything?

**Comment by ralfHielscher** [comment, marked as answer]
Hi Luca,

most likely your MTEX version is too old.

Ralf.

**Reply by lucapbelotti** [reply]
Thanks Ralf. The issue was the older MTEX version.
